tests/bal_functional_behaviour_suite.py

```diff
@@ -0,0 +1,164 @@
+#
+#   Copyright 2013-2023 The Foundry Visionmongers Ltd
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+"""
+A manager test harness test case suite that validates that the
+BasicAssetLibrary manager behaves as expected concerning functional
+behaviors that don't fall under "business-logic", such as performance
+related requirements.
+"""
+
+# pylint: disable=invalid-name, missing-function-docstring, missing-class-docstring
+
+import operator
+import os
+
+from unittest import mock
+
+from openassetio import Context, TraitsData
+from openassetio.exceptions import PluginError
```

Comment:
[fix] Unused import (need to see why the checks aren't catching this)

Reply:
We really need to fix https://github.com/OpenAssetIO/OpenAssetIO-Manager-BAL/actions/runs/5045934836/jobs/9050824142?pr=39#step:5:14

Reply:
https://github.com/OpenAssetIO/OpenAssetIO-Manager-BAL/pull/40

Reply:
Addressed the lints in one commit : https://github.com/OpenAssetIO/OpenAssetIO-Manager-BAL/pull/39#issuecomment-1558195980